### ISSUE TITLE
DNS resolution doesn't work in nspawn container when resolved is disabled

### ISSUE THREAD

**Issue by falconindy** [systemd/systemd#4595]
### Submission type

  - [X] Bug report
  - [ ] Request for enhancement (RFE)

*NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!*

### systemd version the issue has been seen with

v232

*NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!*

### Used distribution

Arch Linux

### In case of bug report: Expected behaviour you didn't see

```
$ getent ahosts rsync.archlinux.org
88.198.91.70    STREAM orion.archlinux.org
88.198.91.70    DGRAM
88.198.91.70    RAW
```

### In case of bug report: Unexpected behaviour you saw

```$ getent ahosts rsync.archlinux.org```
(nothing resolved)

### In case of bug report: Steps to reproduce the problem

```
# systemctl disable --now systemd-resolved
# systemd-nspawn -D /path/to/some/container getent ahosts rsync.archlinux.org
```

Seems related to 3539724c26a1b2b00c4eb3c004b635a4b8647de6. I don't understand the added comment which assumes that based on the `access` check that systemd-resolved is running. It's checking a static file which will be present regardless of whether or not resolved is enabled or running. At the least, it'd be nice to have this behavior configurable via a flag.

I actually can reproduce this behavior even when systemd-resolved is running, but I figure it's better to report the bug based on what I assume to be an error in logic based on the aforementioned comment. It might also be unrelated to this...

**Comment by heftig** [comment]
> I don't understand the added comment which assumes that based on the access check that systemd-resolved is running. It's checking a static file which will be present regardless of whether or not resolved is enabled or running.

I think it assumes RH/Debian-style packaging where `systemd-resolved` (including the above file) is in its own package and automatically enabled+started when installed.

In any case this is a pretty poor check for resolved being available.

**Comment by falconindy** [comment]
I think a more plausible explanation is that it's meant to check the resolv.conf in /run ...

**Comment by heftig** [comment]
Right; that would make more sense, at least for testing whether resolved was started.

You probably still want to know whether the static file exists before attempting to bind it.

**Comment by falconindy** [comment]
This might be a dupe of https://github.com/systemd/systemd/issues/3826. I can see from strace that getent does get a reply, but never considers it.

**Comment by poettering** [comment]
This should be fixed now with #5369 merged.
